hayeon/programmers-1/실패율.js

```diff
@@ -0,0 +1,36 @@
+function solution(N, stages) {
+  const result = [];
+  let totalNum = stages.length;
+
+  for (let i = 1; i <= N; i++) {
+    let failureRate = 0;
+    let stageNum = stages.filter((el) => el === i).length;
+
+    if (stageNum === 0) {
+      failureRate = 0;
+    } else {
+      failureRate = stageNum / totalNum;
+    }
+
+    totalNum -= stageNum;
+    result.push({ idx: i, rate: failureRate });
+  }
+
+  result.sort((a, b) => {
+    if (a.rate === b.rate) {
+      return a.idx - b.idx;
+    } else {
+      return b.rate - a.rate;
+    }
+  });
+  return result.map((el) => el.idx);
+}
```

Comment:
```suggestion
function solution(N, stages) {
  const result = [];
  let totalNum = stages.length;

  for (let i = 1; i <= N; i++) {
    const stageNum = stages.filter((el) => el === i).length;
    const failureRate = stageNum / totalNum;

    totalNum -= stageNum;
    result.push({ stage: i, rate: failureRate });
  }
  
  return result
    .sort((a, b) => b.rate - a.rate || a.stage - b.stage)
    .map((el) => el.stage)
}
```
- 풀이 잘봤습니다 🙇‍♂️
- stageNum은 재할당하는게 없으니 let이 아닌 const로 선언이 옳습니다 (let, const 검토 더 신경쓰시길 부탁드립니다)
- 현재 풀이에서 for문 순회할 때마다 failureRate를 분기 처리로 `0` 또는 `stageNum / totalNum`로 재할당하는 것 보다 그냥 처음에 `const failureRate = stageNum / totalNum`로 선언해서 이를 사용하시면 됩니다.
- 그리고 result에서 프로퍼티 키 네이밍으로 idx보다는 `stage`가 좀 더 좋은 것 같습니다 🙇‍♂️
- sort부분도 저렇게 분기로 return하는 것보다 `||` 연산자를 이용해서 더욱 깔끔한 코드 작성이 가능합니다.

Reply:
sort부분을 좀 더 보기 편하게 작성하려고 해봤는데 || 연산자가 있었군요!!!!! 왜 생각하지 못했을까요,,,😭 변수 선언할 때 더 신경써서 해보겠습니닷 ㅠㅠ